doc/2/guides/main-concepts/3-permissions/index.md

```diff
@@ -235,16 +246,144 @@ Three different profiles can be created using that same role, each with varying
+ - [security:deleteUser](/core/2/api/controllers/security/delete-user/)
+ - [security:getUser](/core/2/api/controllers/security/get-user/)
+ - [security:getUserMapping](/core/2/api/controllers/security/get-user-mapping/)
+ - [security:getUserRights](/core/2/api/controllers/security/get-user-rights/)
+ - [security:mDeleteUsers](/core/2/api/controllers/security/m-delete-users/)
+ - [security:mGetUsers](/core/2/api/controllers/security/m-get-users/)
+ - [security:replaceUsers](/core/2/api/controllers/security/replace-users/)
+ - [security:scrollUsers](/core/2/api/controllers/security/scroll-users/)
+ - [security:searchUsers](/core/2/api/controllers/security/search-users/)
+ - [security:updateUser](/core/2/api/controllers/security/update-user/)
+ - [security:updateUserMapping](/core/2/api/controllers/security/update-user-mapping/)
+
+User credentials related API actions:
+
+ - [security:getCredentials](/core/2/api/controllers/security/get-credentials/)
+ - [security:getCredentialsById](/core/2/api/controllers/security/get-credentials-by-id/)
+ - [security:hasCredentials](/core/2/api/controllers/security/has-credentials/)
+
+
 ## Writing complex or dynamic permission rules
 
 So far, we've seen how to set permissions to API action, using user roles and profiles.
 
-But this is rarely enough to secure an application, as it's commonplace to reject queries or data depending on business rules.
+But this is rarely enough to secure an application, as it's commonplace to **reject queries or data depending on business rules**.
+
 For instance, suppose you have a chat application and you want the users to only be able to edit & delete their own messages: this type of rules cannot be expressed as a simple boolean.
 
 There are multiple ways of adding a business logic layer on top of the standard Kuzzle security one:
 
-* With a [Pipe Plugin](/core/2/plugins/guides/pipes), you can listen to one or multiple [API events](/core/2/plugins/guides/events), and decide whether you accept a query or document according to your business rules (you can see an example on [Github](https://github.com/kuzzleio/kuzzle-plugin-sample-custom-policies))
-* If all you need is to make sure that submitted documents follow a strict set of formatting rules, you can add [document validators](/core/2/guides/cookbooks/datavalidation)
+* With a [Pipe](/core/2/guides/develop-on-kuzzle/3-event-system#pipes), you can listen to one or multiple [API events](/core/2/some-link), and decide whether you accept a query or document according to your business rules (you can see an example on [Github](https://github.com/kuzzleio/kuzzle-plugin-sample-custom-policies))
+* If all you need is to make sure that submitted documents follow a strict set of formatting rules, you can add [document validators](/core/2/guides/advanced/9-data-validation)
+
+::: info
+More information about dynamic rules with pipes: [Event System](/core/2/some-link)
+::: 
+
+## Load permissions
+
+admin controller
```

Comment:
what is this?

Reply:
WIP :sweat_smile:

Reply:
Oh, OK. We should use a comment here, no?

Reply:
(not important)

Reply:
I added the section :+1: